src/main/java/eod/warObject/WarObject.java

```diff
@@ -12,9 +13,11 @@
 public abstract class WarObject implements GameObject {
     public Point position;
     protected Player player;
+    public ArrayList<Class<? extends Event>> canHandle;
```

Comment:
This is a misplace of responsibility. 
Since WarObject doesn't implement EventListener, it should not care about the event it can handle.